tools/infrasys/src/main.rs

```diff
@@ -0,0 +1,448 @@
+            ..Default::default()
+        })
+        .await
+        .unwrap();
+}
+
+/// Adds a BucketPolicy allowing GetObject access to a specified VPC
+/// Input: Region and name of bucket, which folder root.json should be pushed to, and vpcid
+    /// Note that the prefix parameter must have the format "/..." and the bucket name "..."
+/// Behavior: This function will either create the folder that the bucket policy was scoped to, or if the folder exists, simply add to it
+/// Output: Doesn't need to save any metadata from this action
+async fn upload_file(region: &String, bucket_name: &String, prefix: &String, file_path: &String) {
+    let s3_client = S3Client::new(Region::from_str(region).unwrap());
+
+    // File --> Bytes
+    let mut file = File::open(file_path).unwrap();
+    let mut buffer = Vec::new();
+    file.read_to_end(&mut buffer).unwrap();
+
+    let result = s3_client.put_object(PutObjectRequest {
+        bucket: format!("{}{}", bucket_name, prefix),
+        key: "root.json".to_string(),
+        body: Some(StreamingBody::from(buffer)),
+        ..Default::default()
+    }).await.unwrap();
+}
+
+//  =^..^=   =^..^=   =^..^= KEY GENERATION  =^..^=   =^..^=   =^..^=
+
+/// Returns the available keys field of a SigningKeyConfig enum. Available keys represents either all the root keys or publication keys
+fn get_available_keys(signing_key_config: &SigningKeyConfig) -> HashMap<String, String> {
+    match signing_key_config {
+        SigningKeyConfig::file { path } => HashMap::new(),
+        SigningKeyConfig::kms { available_keys, .. } => available_keys.as_ref().unwrap().clone(),
+        SigningKeyConfig::ssm { parameter } => HashMap::new(),
+    }
+}
+
+/// Creates keys using data stored in SigningKeyConfig enum
+/// Output: This method will has an in-place output meaning it edits enum fields with information about resources created
+    /// available-keys starts as a map of pre-existing keyids:regions and will end as a map of pre-existing and generated keyids:regions
+    /// key-stack-arns starts empty and will end as a map of keyids:stackarn if new keys are created
+    /// if key-id is populated, it will not change. Otherwise, it will be populated with a key-id of an available key (TODO)
+async fn create_keys(signing_key_config: &mut SigningKeyConfig) {
+    match signing_key_config {
+        SigningKeyConfig::file { path } => (),
+        SigningKeyConfig::kms {
+            available_keys,
+            regions,
+            key_alias,
+            key_stack_arns,
+            key_id,
+            ..
+        } => {
+            let (available_keys_raw, stack_arns_raw, keyid_raw) = create_kms_keys(
+                key_alias.as_ref(),
+                available_keys.as_ref(),
+                regions.as_ref(),
+            )
+            .await;
+            *available_keys = Some(available_keys_raw);
+            *key_stack_arns = Some(stack_arns_raw);
+            *key_id = Some(keyid_raw);
```

Comment:
I think we can do this in a cleaner way.

If we go with the `KMSConfig` struct that encapsulates all the fields under `kms` I talked about in the comment in `pubsys-config`, we can `impl` `create_kms_keys` as one of its methods. The method could just modify `self` and it becomes much clearer.

The function signature could look like this:
```
impl KMSConfig {
    fn create_kms_key(&mut self) -> Result<()> {
        //Create the kms keys
        ...
        // Set the fields accordingly
        self.available_keys = ...    
        ..
        Ok(())
   }
}
```

Then you can just call it like so here:
```suggestion
            kms_config.create_kms_keys().context(error::CreateKMSKeys)?;
```